src/Model/User.php

```diff
@@ -78,21 +78,105 @@ public function __construct( \WP_User $user ) {
 	 */
 	protected function is_private() {
 
+		/**
+		 * Filter whether the user is private.
+		 *
+		 * If true, the user will be considered private without further checks.
+		 *
+		 * @param mixed|null|boolean $is_private If set, return without executing the manual query
+		 *                                       to see if the user has published posts.
+		 * @param User               $this       Instance of the User Model
+		 */
+		$is_private = apply_filters( 'graphql_user_model_is_private', null, $this );
```

Comment:
This is kind of already handled here -> https://github.com/wp-graphql/wp-graphql/blob/develop/src/Model/Model.php#L222 though the query will still run, so maybe we just need to rework that lower level filter, but I think we should keep all of the Model Layer hooks & filters at the abstract level so they are consistent across all types.

Reply:
@CodeProKid If we were to use the existing `graphql_data_is_private` filter, it _still_ executes this `is_private` method, which will execute the query to check for published posts, even if a filter were applied to override the results.

Perhaps we could do what I'm doing here in the Model.php file though, to centralize it better. 

Basically, I want to provide an "escape hatch" for folks to _completely_ override the automated checks we're doing on their behalf. 

For example, if you just have a hard rule that ALL USERS should be considered private, always, then you could avoid the lookup for published posts altogether and just return true with no overhead. 

If you were to do that with the current filter, and just `return__true` it still executes the query. 

I didn't think it would, but upon testing and actually monitoring, it indeed does. 🤷‍♂

Reply:
yeah, that's what I'm proposing. Move the existing filter in Model.php to before we call `$this->is_private();` and only call `$this->is_private();` if the filter is empty, otherwise use the results of the filter.

Reply:
:+1: we would use this filter and it would solve the entire perf problem for us